lib/shopify_api/api_version.rb

```diff
@@ -1,113 +1,111 @@
 # frozen_string_literal: true
 module ShopifyAPI
-  class ApiVersion
+  class ApiVersion < ActiveResource::Base
     class UnknownVersion < StandardError; end
     class InvalidVersion < StandardError; end
-
-    extend DefinedVersions
-
     include Comparable
 
-    def self.coerce_to_version(version_or_name)
-      return version_or_name if version_or_name.is_a?(ApiVersion)
-
-      @versions ||= {}
-      @versions.fetch(version_or_name.to_s) do
-        raise UnknownVersion, "#{version_or_name} is not in the defined version set: #{@versions.keys.join(', ')}"
+    self.site = "https://app.shopify.com/services/"
+    self.primary_key = :handle
+
+    HANDLE_FORMAT = /^\d{4}-\d{2}$/.freeze
+    UNSTABLE_HANDLE = 'unstable'
+    UNSTABLE_AS_DATE = Time.utc(3000, 1, 1)
+    API_PREFIX = '/admin/api/'
+
+    class << self
+      attr_reader :versions
+
+      def coerce_to_version(version_or_handle)
+        return version_or_handle if version_or_handle.is_a?(ApiVersion)
+        if @versions.nil?
+          warn "[API VERSION WARNING] Known API Version set is empty. Initializing unvalidated version from handle."
+          return ApiVersion.new(handle: version_or_handle)
+        end
+
+        @versions[version_or_handle.to_s].tap do |api_version|
+          unless api_version
+            raise ApiVersion::UnknownVersion, "UnknownVersion API version specified, `#{version_or_handle}`. \n Versions available: #{@versions.keys}"
+          end
+        end
       end
-    end
 
-    def self.define_version(version)
-      @versions ||= {}
+      def define_known_versions
+        @versions = Meta.admin_versions.map { |version| [version.handle, version] }.to_h
+      rescue ActiveResource::ConnectionError => e
+        warn "[API VERSION WARNING] Could not fetch Admin API versions."
+        warn "[API VERSION WARNING] #{e.message}"
+        @versions = nil
+      end
 
-      @versions[version.name] = version
-    end
+      def clear_defined_versions
+        @versions = nil
+      end
 
-    def self.clear_defined_versions
-      @versions = {}
+      def latest_stable_version
+        warn(
+          '[DEPRECATED] ShopifyAPI::ApiVersion.latest_stable_version is deprecated and will be removed in a future version.'
```

Comment:
Is there a recommended way to do this same thing after this method is removed?  Is this just discouraged going forward?

Reply:
I think we wanted to discourage anyone trying to set the version to the latest version automatically since it's likely to break apps. Bumping to the latest version should hopefully be intentional. 

It can still be computed however since each ApiVersion has a `latest_supported` attribute, its just a little more work.

`ShopifyAPI::Meta.admin_versions.find(&:latest_supported?)` works at the expense of another api call, but we could also cache the versions in `Meta.admin_versions`

Reply:
latest_supported_version is used by the ShopifyApp gem https://github.com/Shopify/shopify_app/blob/0900ddcb2f268f6efe3e184739ee83eeab3a0020/lib/generators/shopify_app/install/install_generator.rb#L12 to run the generators.   

It can now use `ShopifyAPI::Meta.admin_versions.find(&:latest_supported?)` to find that info.